cms/bundles/viewsets.py

```diff
@@ -0,0 +1,308 @@
+                    action = "bundles.approve"
+                    kwargs["data"] = {"old": original_status}
+                    notify_slack_of_status_change(instance, original_status, user=self.request.user, url=url)
+                elif instance.status == BundleStatus.RELEASED.value:
+                    action = "wagtail.publish"
+                    self.start_time = time.time()
+                else:
+                    action = "bundles.update_status"
+                    kwargs["data"] = {
+                        "old": original_status,
+                        "new": instance.get_status_display(),
+                    }
+                    notify_slack_of_status_change(instance, original_status, user=self.request.user, url=url)
+
+                # now log the status change
+                log(
+                    action=action,
+                    instance=instance,
+                    **kwargs,
+                )
+
+        return instance
+
+    def run_after_hook(self) -> None:
+        """This method allows calling hooks or additional logic after an action has been executed.
+
+        In our case, we want to send a Slack notification if manually published, and approve any of the
+        related pages that are in a Wagtail workflow.
+        """
+        if self.action == "publish" or (self.action == "edit" and self.object.status == BundleStatus.RELEASED):
+            notify_slack_of_publication_start(self.object, user=self.request.user)
+            start_time = self.start_time or time.time()
+            for page in self.object.get_bundled_pages():
+                if page.current_workflow_state:
+                    page.current_workflow_state.current_task_state.approve(user=self.request.user)
+
+            notify_slack_of_publish_end(self.object, time.time() - start_time, user=self.request.user)
+
+    def get_context_data(self, **kwargs: Any) -> dict:
+        """Updates the template context.
+
+        Show the "save and approve" button if the bundle has the right status, and we have a different user
+        than the creator
+        """
+        context: dict = super().get_context_data(**kwargs)
+
+        context["show_save_and_approve"] = (
+            self.object.can_be_approved and self.form.for_user.pk != self.object.created_by_id
+        )
+        context["show_publish"] = (
+            self.object.status == BundleStatus.APPROVED and not self.object.scheduled_publication_date
+        )
+
+        return context
+
+
+class BundleInspectView(InspectView):
+    """The Bundle inspect view class."""
+
+    template_name = "bundles/wagtailadmin/inspect.html"
+
+    def get_fields(self) -> list[str]:
+        """Returns the list of fields to include in the inspect view."""
+        return ["name", "status", "created_at", "created_by", "approved", "scheduled_publication", "pages"]
+
+    def get_field_label(self, field_name: str, field: "Field") -> Any:
```

Comment:
Type ignore last line if needed. But return type should be predicable to callers.

```suggestion
    def get_field_label(self, field_name: str, field: "Field") -> str:
```